django/utils/timesince.py

```diff
@@ -69,22 +71,38 @@ def timesince(d, now=None, reversed=False, time_strings=None, depth=2):
             leapdays += 1
     delta -= datetime.timedelta(leapdays)
 
+    # fix months duration to be relative to the months between d and now
+    months = now.month - d.month
+    if months > 0:
+        m_days = (sum(MONTHS_DAYS[d.month - 1 : now.month - 1])) / months
+    else:
+        m_days = (365 - sum(MONTHS_DAYS[now.month - 1 : d.month - 1])) / (12 + months)
```

Comment:
Can you extract this into a separate helper function, so that its purpose is clearer? Maybe a docstring? And a few unit tests.